nixos/modules/services/misc/recyclarr.nix

```diff
@@ -0,0 +1,166 @@
+{
+  config,
+  lib,
+  pkgs,
+  utils,
+  ...
+}:
+
+let
+  cfg = config.services.recyclarr;
+  format = pkgs.formats.yaml { };
+  configPath = "/var/lib/recyclarr/config.json";
```

Comment:
See, this is where I'm wondering whether having `cfg.workingDir` is useful: it makes handling the configuration with secrets more compilcated (where do we store it, in `workingDir` or in a `StateDirectory`? What if they're the same? etc...)

At the very least, given that you don't have a `StateDirectory` in your service, this path isn't guaranteed to exist.

IMO I'd just remove `workingDir`, use `StateDirectory = "recyclarr"` (which allows us to remove the `tempfiles` rule) and tell it to use `/var/lib/recyclarr`.

If you have strong feelings against it, please at least handle the configuration path correctly for non-default `workingDir` values.

Reply:
Removed `workingDir` in favor of `StateDirectory` approach.